### PR TITLE
Fix initial points sampling

### PR DESCRIPTION
**Pull request by thomasckng**
Fix bug in `jim.sample()` and use the same code for `HeterodynedTransientLikelihoodFD.maximize_likelihood()`.

### CODE REVIEW

**Comment by thomasckng** [comment]
Replace #153

File: src/jimgw/jim.py [hunk 1]
```diff
@@ -104,19 +104,30 @@ def posterior(self, params: Float[Array, " n_dim"], data: dict):
...
+                    ~jax.tree.reduce(
+                        jnp.logical_and,
+                        jax.tree.map(lambda x: jnp.isfinite(x), initial_position),
+                    ),
+                    axis=1,
+                )
 
                 key, subkey = jax.random.split(key)
-                guess = self.prior.sample(subkey, self.sampler.n_chains)
+                guess = self.prior.sample(subkey, non_finite_index.sum())
```
Comment (kazewong) [inline src/jimgw/jim.py:124]: The guesses are not guaranteed to be finite, hence the original implementation. Please change it back

File: src/jimgw/single_event/likelihood.py [hunk 2]
```diff
@@ -558,14 +558,14 @@ def maximize_likelihood(
     ):
         parameter_names = prior.parameter_names
         for transform in sample_transforms:
-            parameter_names = transform.propagate_name(parameter_names)
+            parameter_names = jax.vmap(transform.propagate_name)(parameter_names)
```
Comment (kazewong) [inline src/jimgw/single_event/likelihood.py:561]: One does not need vmap for propagating the names

**Comment by thomasckng** [comment]
Close this for a moment

**Pull request closed without merging**